aws/rust-runtime/aws-config/src/profile/parser.rs

```diff
@@ -143,35 +167,151 @@ impl ProfileSet {
         Self {
             profiles: Default::default(),
             selected_profile: "default".into(),
+            sso_sessions: Default::default(),
         }
     }
 }
 
-/// An individual configuration profile
-///
-/// An AWS config may be composed of a multiple named profiles within a [`ProfileSet`].
+/// Represents a top-level section (e.g., `[profile name]`) in a config file.
+pub(crate) trait Section {
+    /// The name of this section
+    fn name(&self) -> &str;
+
+    /// Returns all the properties in this section
+    fn properties(&self) -> &HashMap<String, Property>;
+
+    /// Returns a reference to the property named `name`
+    fn get(&self, name: &str) -> Option<&str>;
+
+    /// True if there are no properties in this section.
+    fn is_empty(&self) -> bool;
+
+    /// Insert a property into a section
+    fn insert(&mut self, name: String, value: Property);
+}
+
 #[derive(Debug, Clone, Eq, PartialEq)]
-pub struct Profile {
+struct SectionInner {
     name: String,
     properties: HashMap<String, Property>,
 }
 
+impl Section for SectionInner {
+    fn name(&self) -> &str {
+        &self.name
+    }
+
+    fn properties(&self) -> &HashMap<String, Property> {
+        &self.properties
+    }
+
+    fn get(&self, name: &str) -> Option<&str> {
+        self.properties
+            .get(to_ascii_lowercase(name).as_ref())
+            .map(|prop| prop.value())
+    }
+
+    fn is_empty(&self) -> bool {
+        self.properties.is_empty()
+    }
+
+    fn insert(&mut self, name: String, value: Property) {
+        self.properties
+            .insert(to_ascii_lowercase(&name).into(), value);
+    }
+}
+
+/// An individual configuration profile
+///
+/// An AWS config may be composed of a multiple named profiles within a [`ProfileSet`].
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Profile(SectionInner);
+
 impl Profile {
     /// Create a new profile
-    pub fn new(name: String, properties: HashMap<String, Property>) -> Self {
-        Self { name, properties }
+    pub fn new(name: impl Into<String>, properties: HashMap<String, Property>) -> Self {
+        Self(SectionInner {
+            name: name.into(),
+            properties,
+        })
     }
 
     /// The name of this profile
     pub fn name(&self) -> &str {
-        &self.name
+        self.0.name()
     }
 
     /// Returns a reference to the property named `name`
     pub fn get(&self, name: &str) -> Option<&str> {
-        self.properties
-            .get(to_ascii_lowercase(name).as_ref())
-            .map(|prop| prop.value())
+        self.0.get(name)
+    }
```

Comment:
Yeah, they are. However, removing them will be a breaking change since importing `Section` would be required. I added them to `SsoSession` to be consistent.